adapters/oidc/js/src/main/resources/keycloak.d.ts

```diff
@@ -37,6 +37,33 @@ declare namespace Keycloak {
 	type KeycloakPromiseType = 'legacy' | 'native';
 	type KeycloakPkceMethod = 'S256';
 
+	interface KeycloakCredentials {
+		/**
+		 * Secret or Signed JWT.
+		 */
+		secret: string;
+	}
+
+	interface KeycloakConfig {
+		/**
+		 * URL to the Keycloak server, for example: http://keycloak-server/auth
+		 */
+		url?: string;
+		/**
+		 * Name of the realm, for example: 'myrealm'
+		 */
+		realm: string;
+		/**
+		 * Client identifier, example: 'myapp'
+		 */
+		clientId: string;
+		/**
+		 * The credentials object contains the secret property that should be used
+		 * depending on which flow and access type was chosen.
+		 */
+		credentials?: KeycloakCredentials;

```

Comment:
@jonkoops overall looks good to me. I just wonder if we really need a separate interface just to store secrets, instead of just use `credentials?: string;`

@ssilvert what do you think?

Reply:
Could be an inline thing, in this case I am just adding it to ensure that you could also use this interface separately as a user.

Reply:
In documentation we have this line: 
````
To use the JavaScript adapter you must first create a client for your application in the Keycloak Administration Console. Make sure public is selected for Access Type.
````

Clients which are used with Javascript adapter should be public which means they don't have credentials. So should this be removed or I missed something?

Reply:
@mhajas If that is indeed the case, should this entire functionality not be removed from the JavaScript Adapter in the first place?

Reply:
Since KeycloakCredentials just wraps a string I don't see how it adds any value.

The rest looks good.

Reply:
@jonkoops this discussion is probably out of scope for this PR, but it seems useless to me. I will start a thread in dev ML.